### PR TITLE
Add details on how to run dockerised dependencies

### PR DESCRIPTION
**Pull request by nickstenning**
Some more deployment docs.

### CODE REVIEW

File: docs/INSTALL.rst [hunk 1]
```diff
@@ -48,13 +48,44 @@ This will take some time to complete, as it will install and configure a
...
+-  a PostgreSQL database
+-  an ElasticSearch server with the ICU analysis plugin enabled
+-  an nsqd server
+-  a Redis server
+-  a mailer
+
+One option is to containerise these services, although you should investigate
+for yourself if this is a sensible approach for your environment.
+
+For example. We choose to run ElasticSearch, nsqd, and Redis in containers. We
```
Comment (tilgovi) [inline docs/INSTALL.rst]: "For example,"

File: docs/INSTALL.rst [hunk 2]
```diff
@@ -48,13 +48,44 @@ This will take some time to complete, as it will install and configure a
...
+also use an `ambassador container`_ to point to our mailserver::
+
+    $ docker run -d --name elasticsearch nickstenning/elasticsearch-icu
+    $ docker run -d --name nsqd --expose 4150 --expose 4151 nsqio/nsq /nsqd
+    $ docker run -d --name redis redis
+    $ docker run -d --name mail --expose 25 -e MAIL_PORT_25_TCP=tcp://smtp.mydomain.com:25 svendowideit/ambassador
```
Comment (tilgovi) [inline docs/INSTALL.rst]: I'm not sure hostnames work here. I would put an IP in the example for the ambassador.

File: docs/INSTALL.rst (hunk 1 above)
Comment (nickstenning) [inline docs/INSTALL.rst]: English: how does it work? Fixed.

File: docs/INSTALL.rst (hunk 2 above)
Comment (nickstenning) [inline docs/INSTALL.rst]: It's just `socat(1)` under the hood. Hostnames work fine.

Comment (nickstenning) [inline docs/INSTALL.rst]: (I have actually checked this. I'm not all bluff and bluster, I promise.)

Comment (renoirb) [inline docs/INSTALL.rst]: Uhm, right.

After reading the notes about it in phusion/baseimage, I thought to add the following lines in the Dockerfile

````
ENV STATSD_PORT_8125_UDP $DOCKER_STATSD_PORT_8125_UDP
ENV ELASTICSEARCH_INDEX $DOCKER_ELASTICSEARCH_INDEX
ENV ELASTICSEARCH_PORT_9200_TCP_ADDR $DOCKER_ELASTICSEARCH_PORT_9200_TCP_ADDR
ENV ELASTICSEARCH_PORT_9200_TCP_PORT $DOCKER_ELASTICSEARCH_PORT_9200_TCP_PORT
ENV MAIL_DEFAULT_SENDER $DOCKER_MAIL_DEFAULT_SENDER
ENV MAIL_PORT_25_TCP_ADDR $DOCKER_MAIL_PORT_25_TCP_ADDR
ENV MAIL_PORT_25_TCP_PORT $DOCKER_MAIL_PORT_25_TCP_PORT
ENV NSQD_PORT_4150_TCP_ADDR $DOCKER_NSQD_PORT_4150_TCP_ADDR
ENV NSQD_PORT_4150_TCP_PORT $DOCKER_NSQD_PORT_4150_TCP_PORT
ENV NSQD_PORT_4151_TCP_ADDR $DOCKER_NSQD_PORT_4151_TCP_ADDR
ENV NSQD_PORT_4151_TCP_PORT $DOCKER_NSQD_PORT_4151_TCP_PORT
ENV NSQ_NAMESPACE $DOCKER_NSQ_NAMESPACE
ENV REDIS_PORT_6379_TCP_ADDR $DOCKER_REDIS_PORT_6379_TCP_ADDR
ENV REDIS_PORT_6379_TCP_PORT $DOCKER_REDIS_PORT_6379_TCP_PORT
ENV CLIENT_SECRET $DOCKER_CLIENT_SECRET
ENV SECRET_KEY $DOCKER_SECRET_KEY

RUN echo $ELASTICSEARCH_PORT_9200_TCP_ADDR
```

Then before running with `docker build`/`docker run` I source a file that has the current live variables.  

What do you think?

Comment (tilgovi) [inline docs/INSTALL.rst]: Don't modify the Dockerfile. Read the docs.
`docker run --env-file`

Comment (tilgovi) [inline docs/INSTALL.rst]: @nickstenning I remember it not working for me, but I looked at the socat docs and you seem to be right.

Comment (renoirb) [inline docs/INSTALL.rst]: OK @tilgovi, but which docs? What/where should I read about how to use `--env-file`. Even the updated docs doesn’t mention it.

Comment (tilgovi) [inline docs/INSTALL.rst]: Docker docs.

**Pull request merged**